server/src/main/java/org/opensearch/monitor/fs/FsProbe.java

```diff
@@ -206,6 +206,9 @@ public static FsInfo.Path getFSInfo(NodePath nodePath) throws IOException {
         fsPath.free = adjustForHugeFilesystems(nodePath.fileStore.getUnallocatedSpace());
         fsPath.available = adjustForHugeFilesystems(nodePath.fileStore.getUsableSpace());
         fsPath.fileCacheReserved = adjustForHugeFilesystems(nodePath.fileCacheReservedSize.getBytes());
+        // fsPath.fileCacheUtilized = adjustForHugeFilesystems(...);
```

Comment:
I think we could just leave comment here:

```suggestion
        // The fileCacheUtilize could not be computed here because information about utilization of file cache is hold by FileCache which is not accessible here and since this method is static we can not assume relevant context.
```

Reply:
@reta Sure, I will do something like that. Again, the comment I left here was mostly for reviewers to explain my approach.

It is out of the scope of this PR but in general I am not happy about `static getFSInfo(NodePath nodePath)` method at all. IMO when file caches were introduced this method and its use should have been redesigned. The fact that file cache utilization value requires a file cache instance is not surprising but then why there is such `static` method that can give you `FsInfo.Path` instance except the file cache utilization value is not calculated at all. IMO it leads to errors (and I believe that is what I was fighting with the most while adding various tests for file caches while trying to use conventional default initialization values, ie. `-1` for variables, this was leading to various edge cases and unexpected test failures).

BTW: Notice how the value of file cache utilization is actually delivered. Excerpt from `public FsInfo stats(FsInfo previous)` (which is in `FsProbe` class):

```java
        NodePath[] dataLocations = nodeEnv.nodePaths();
        FsInfo.Path[] paths = new FsInfo.Path[dataLocations.length];
        for (int i = 0; i < dataLocations.length; i++) {
            paths[i] = getFSInfo(dataLocations[i]);
            if (fileCache != null && dataLocations[i].fileCacheReservedSize.compareTo(ByteSizeValue.ZERO) >= 0) {
                paths[i].fileCacheReserved = adjustForHugeFilesystems(dataLocations[i].fileCacheReservedSize.getBytes());
                paths[i].fileCacheUtilized = adjustForHugeFilesystems(fileCache.usage().usage());
                paths[i].available -= (paths[i].fileCacheReserved - paths[i].fileCacheUtilized);
            }
        }
```
The instance of `FsProbe` that "hosts" this method requires `FileCache` in the ctor, so the `fileCache` is actually available, but still the code first calls the `static` method (line no.4) which is not able to calculate file cache utilization and then it updates the value using the cache instance (line no.7).

After I studied history of this file I can see how this happened but I believe it is time to review the use of `FsProbe.getFSInfo(...)` all over the code.

Reply:
Sure, please suggest better solution

Reply:
Finally, addressing this design issue. In fact this issue has been the reason why tests were failing. Please see `FsProbeTests.testAdjustForHugeFilesystems()`.

It seems that there were some cases where the `FsInfo.Path` instances were created using the `static FsInfo.Path#getFSInfo(NodePath nodePath)` and hence the file cache utilization was `-1`.

After the recent PR update this should be fixed but we need to be extra careful about the logic in `FsInfo.stats(FsInfo previous)` method. Specifically the `fileCache` used. My understanding is that the `fileCache` is always attached tot he first node data path only, see `NodeEnvironment` ctor where it reads:

```java
this.fileCacheNodePath = nodePaths[0];
```

I will keep investigating.